go/testimonyd/internal/socket/socket.c

```diff
@@ -28,6 +28,26 @@
 #define UNIX_PATH_MAX 108
 #endif
 
+// Function for waiting for new blocks using select() function
+int WaitForBlocks(int sock_fd) {
+  struct timeval tv;
+  fd_set fds;
+  int res;
+
+  FD_ZERO(&fds);
+  FD_SET(sock_fd, &fds);
+
+  tv.tv_sec = 0;
+  tv.tv_usec = 10000;
```

Comment:
since this should unblock on receipt, make this sleep longer (1 sec, maybe?)

Reply:
changed to 1 second